include/clap/ext/draft/mini-curve-display.h

```diff
@@ -5,19 +5,87 @@
+
+   // The mini curve is intended to draw the transfer curve of the plugin.
+   // In this case the both x and y values are in dB.
+   // This would be useful in for example a compressor or distortion plugin.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_TRANSFER_CURVE = 3,
+
+   // This mini curve is intended to draw gain reduction over time. In this case
+   // x refers to the window in seconds and y refers to level in dB, x_min is
+   // always 0, and x_max would be the duration of the window.
+   // This would be useful in for example a compressor or limiter.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_GAIN_REDUCTION = 4,
+
+   // This curve is intended as a generic time series plot. In this case
+   // x refers to the window in seconds. x_min is always 0, and x_max would be the duration of the
+   // window.
+   // Y is not specified and up to the plugin.
+   CLAP_MINI_CURVE_DISPLAY_CURVE_KIND_TIME_SERIES = 5,
+
+   // Note: more entries could be added here in the future
+};
+
+typedef struct clap_mini_display_curve_hints {
+
+   // Range for the x axis.
+   double x_min;
+   double x_max;
+
+   // Range for the y axis.
+   double y_min;
+   double y_max;
+
+} clap_mini_display_curve_hints_t;
+
+// A set of points representing the curve to be painted.
+typedef struct clap_mini_display_curve_data {
+   // Indicates the kind of curve those values represent, the host can use this
+   // information to paint the curve using a meaningful color.
+   int32_t curve_kind;
+
+   // values[0] will be the leftmost value and values[data_size -1] will be the rightmost
+   // value.
+   //
+   // The value 0 and UINT16_MAX won't be painted.
+   // The value 1 will be at the bottom of the curve and UINT16_MAX - 1 will be at the top.
+   uint16_t *values;
+   uint32_t  values_count;
+} clap_mini_display_curve_data_t;
+
 typedef struct clap_plugin_mini_curve_display {
+   // Returns the number of curves the plugin wants to paint.
+   // Be aware that the space to display those curves will be small, and too much data will make
+   // the output hard to read.
+   uint32_t(CLAP_ABI *get_curve_count)(const clap_plugin_t *plugin);
+
    // Renders the curve into the data buffer.
-   // The value 0 will be at the bottom of the curve and UINT16_MAX will be at the top.
-   // The value at index 0 will be the leftmost and the value at index data_size -1 will be the
-   // rightmost.
+   //
+   // The host will "stack" the curves, from the first one to the last one.
+   // data[0] is the first curve to be painted.
+   // data[n + 1] will be painted over data[n].
+   //
+   // Returns the number of curves rendered.
    // [main-thread]
-   bool(CLAP_ABI *render)(const clap_plugin_t *plugin, uint16_t *data, uint32_t data_size);
+   uint32_t(CLAP_ABI *render)(const clap_plugin_t            *plugin,
+                              clap_mini_display_curve_data_t *data,
+                              uint32_t                        data_size);
```

Comment:
Is data size the size of each curve or af all curves? That is if I return 3 curves of size 256 do I -it 256 or 768 in data size. Is data size the number of elements or the size in bytes also?

I think a comment explaining this and also renaming to “Curve_Point_Count” or some such may help a lot now there are multi curve anilities (Which are cool)

Reply:
I could name those `curves` and `curves_capacity`.